http://bugzilla.xfce.org/show_bug.cgi?id=10923

From 1bc7f32ea8244575124420bd34d52913831a1ae1 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Fri, 30 May 2014 19:12:02 +0300
Subject: [PATCH] Update for Upower 0.99

Upower 0.99 changed from the "changed" signal to "notify" as well as
the function signature. This patch adds that while still supporting
the previous versions of upower.
---
 panel-plugin/systemload.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/panel-plugin/systemload.c b/panel-plugin/systemload.c
index c0572a3..30636e9 100644
--- a/panel-plugin/systemload.c
+++ b/panel-plugin/systemload.c
@@ -624,11 +624,15 @@ monitor_set_orientation (XfcePanelPlugin *plugin, GtkOrientation orientation,
 
 #ifdef HAVE_UPOWER_GLIB
 static void
+#if UP_CHECK_VERSION(0, 99, 0)
+upower_changed_cb(UpClient *client, GParamSpec *pspec, t_global_monitor *global)
+#else /* UP_CHECK_VERSION < 0.99 */
 upower_changed_cb(UpClient *client, t_global_monitor *global)
+#endif /* UP_CHECK_VERSION */
 {
     setup_timer(global);
 }
-#endif
+#endif /* HAVE_UPOWER_GLIB */
 
 static void
 entry_changed_cb(GtkEntry *entry, t_global_monitor *global)
@@ -925,10 +929,15 @@ systemload_construct (XfcePanelPlugin *plugin)
 
 #ifdef HAVE_UPOWER_GLIB
     if (global->upower) {
+#if UP_CHECK_VERSION(0, 99, 0)
+        g_signal_connect (global->upower, "notify",
+                          G_CALLBACK(upower_changed_cb), global);
+#else /* UP_CHECK_VERSION < 0.99 */
         g_signal_connect (global->upower, "changed",
                           G_CALLBACK(upower_changed_cb), global);
+#endif /* UP_CHECK_VERSION */
     }
-#endif
+#endif /* HAVE_UPOWER_GLIB */
     
     g_signal_connect (plugin, "free-data", G_CALLBACK (monitor_free), global);
 
-- 
1.9.3

